File: app/src/main/java/com/rubyhuntersky/peregrine/model/FundingOption.java
package com.rubyhuntersky.peregrine.model;

import android.os.Parcelable;

import java.math.BigDecimal;

/**
 * @author wehjin
 * @since 2/6/16.
 */
public interface FundingOption extends Parcelable {
    String getAssetName();
    BigDecimal getSharesAvailableToSell();
    BigDecimal getSellPrice();
    BigDecimal getValueWhenSold();
}

File: app/src/main/java/com/rubyhuntersky/peregrine/model/FundingOption.kt
package com.rubyhuntersky.peregrine.model

import android.os.Parcelable

import java.math.BigDecimal

interface FundingOption : Parcelable {
    val assetName: String
    val sharesAvailableToSell: BigDecimal
    val sellPrice: BigDecimal
    val valueWhenSold: BigDecimal

    fun getSharesToRaiseFunds(funds: BigDecimal): BigDecimal {
        return if (sellPrice == BigDecimal.ZERO) {
            BigDecimal.ZERO
        } else {
            funds.divide(sellPrice, Values.SCALE, BigDecimal.ROUND_HALF_UP)
                    .setScale(0, BigDecimal.ROUND_CEILING)
        }
    }
}
